ConfigSpace/hyperparameters.pyx

```diff
@@ -877,6 +998,327 @@ cdef class NormalFloatHyperparameter(FloatHyperparameter):
+            unless it is NotImplemented.
+
+        """
+        if not isinstance(other, self.__class__):
+            return False
+
+        return (
+            self.name == other.name and
+            self.default_value == other.default_value and
+            self.alpha == other.alpha and
+            self.beta == other.beta and
+            self.log == other.log and
+            self.q == other.q and
+            self.lower == other.lower and
+            self.upper == other.upper
+        )
+
+    def __copy__(self):
+        return BetaFloatHyperparameter(
+            name=self.name,
+            default_value=self.default_value,
+            alpha=self.alpha,
+            beta=self.beta,
+            log=self.log,
+            q=self.q,
+            lower=self.lower,
+            upper=self.upper,
+            meta=self.meta
+        )
+
+    def __hash__(self):
+        return hash((self.name, self.alpha, self.beta, self.log, self.q))
+
+    def to_uniform(self) -> 'UniformFloatHyperparameter':
+        lb = self.lower
+        ub = self.upper
+
+        return UniformFloatHyperparameter(self.name,
+                                          lb,
+                                          ub,
+                                          default_value=self.default_value,
+                                          q=self.q, log=self.log)
+
+    def check_default(self, default_value: Union[int, float]) -> Union[int, float]:
+        # return mode as default
+        lb = self.lower
+        ub = self.upper
+        
+        if (default_value is None) or (default_value < lb) or (default_value > ub):
+            if (self.alpha > 1) and (self.beta > 1):
+                normalized_mode = (self.alpha - 1) / (self.alpha + self.beta - 2)
+                return self._transform_scalar((self._upper - self._lower) * normalized_mode + self._lower)
+            else: 
+                return (ub - lb) / 2
+
+        elif self.is_legal(default_value):
+            return default_value
+        else:
+            raise ValueError("Illegal default value %s" % str(default_value))
+
+    def to_integer(self) -> 'BetaIntegerHyperparameter':
+        if self.q is None:
+            q_int = None
+        else:
+            q_int = int(self.q)
+        return BetaIntegerHyperparameter(self.name, lower=self.lower, upper=self.upper, alpha=self.alpha, beta=self.beta,
+                                           default_value=int(np.round(self.default_value, 0)),
+                                           q=q_int, log=self.log)
+
+    def is_legal(self, value: Union[float]) -> bool:
+        if not (isinstance(value, float) or isinstance(value, int)):
+            return False
+        elif self.upper >= value >= self.lower:
+            return True
+        else:
+            return False
+
+    cpdef bint is_legal_vector(self, DTYPE_t value):
+        if self.upper >= value >= self.lower:
```

Comment:
Likewise here?

Reply:
Same answer =)

Reply:
@eddiebergman I was wrong here, but it is now fixed!